tests/bluetooth/host/mock_host/src/net/buf.c

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2022 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/zephyr.h>
+#include <zephyr/net/buf.h>
+#include <zephyr/bluetooth/buf.h>
+#include "buf_help_utils.h"
+
+
+static uint8_t *fixed_data_alloc(struct net_buf *buf, size_t *size,
+			      k_timeout_t timeout)
+{
+	ztest_test_fail();
+	return NULL;
+}
+
+static void fixed_data_unref(struct net_buf *buf, uint8_t *data)
+{
+	/* Nothing needed for fixed-size data pools */
+	ztest_test_fail();
+}
+
+const struct net_buf_data_cb net_buf_fixed_cb = {
+	.alloc = fixed_data_alloc,
+	.unref = fixed_data_unref,
+};
+
+struct net_buf *net_buf_alloc_fixed(struct net_buf_pool *pool, k_timeout_t timeout)
+{
+	ztest_check_expected_value(pool);
+	utils_net_buf_alloc_timeout_value_mock(timeout.ticks);
```

Comment:
thinking a bit more about it, my opinion is that this function should be defined here, since it's essentially a mock: It only exists to workaround a limitation of the testing framework, but it is checking parameter values of `net_buf_alloc_fixed`. 

You can keep the declaration in `buf_help_utils.h` though.